packages/kit/src/utils/escape.js

```diff
@@ -24,54 +24,43 @@ export function escape_json_in_html(val) {
 	);
 }
 
-/**
- * @param str {string} string to escape
- * @param dict {Record<string, string>} dictionary of character replacements
- * @param unicode_encoder {function(number): string} encoder to use for high unicode characters
- * @returns {string}
- */
-function escape(str, dict, unicode_encoder) {
-	let result = '';
-
-	for (let i = 0; i < str.length; i += 1) {
-		const char = str.charAt(i);
-		const code = char.charCodeAt(0);
-
-		if (char in dict) {
-			result += dict[char];
-		} else if (code >= 0xd800 && code <= 0xdfff) {
-			const next = str.charCodeAt(i + 1);
-
-			// If this is the beginning of a [high, low] surrogate pair,
-			// add the next two characters, otherwise escape
-			if (code <= 0xdbff && next >= 0xdc00 && next <= 0xdfff) {
-				result += char + str[++i];
-			} else {
-				result += unicode_encoder(code);
-			}
-		} else {
-			result += char;
-		}
-	}
-
-	return result;
-}
-
 /** @type {Record<string, string>} */
 const escape_html_attr_dict = {
-	'<': '&lt;',
-	'>': '&gt;',
+	'&': '&amp;',
 	'"': '&quot;'
 };
 
+const escape_html_attr_regex = new RegExp(
```

Comment:
can we add a comment about why this is needed?

Reply:
Why delete < and >
they break HTML if not escaped?

Reply:
and we fixed it with #3798 and #3804

Reply:
@PH4NTOMiki That was about `escape_json_in_html`, this is about `escape_html_attr`.

@benmccann Sorry about the comments, the previous code didn't have them and I thought they weren't necessary. I'll add some.